pkg/aws/awscontrolplane/admit_awscontrolplane_test.go

```diff
@@ -90,13 +103,28 @@ func TestAWSControlPlaneAdmit(t *testing.T) {
 			g8sControlplaneReplicas: nil,
 			expectAvailabilityZones: nil,
 			validAvailabilityZones:  []string{"eu-central-1a", "eu-central-1b", "eu-central-1c"},
+			currentInstanceType:     "m4.xlarge",
+			expectedInstanceType:    "",
+		},
+		{
+			// Here we check if the instance type is defaulted
```

Comment:
As input for future design decisions, when you want to test instance type defaulting it would be good to have a dedicated test that does not have to care about AZs, in case such information are irrelevant for the cause. The implication would then be that the design of your defaulting functionality has to support the separate execution and verification. You get 3 different objectives mixed in one "blackbox" of code and maintaining the codebase and ensuring testability will become a nightmare. We had the same issue in the `api` and started to isolate defaulting and validation on a case by case basis.

Reply:
Yeah I agree we better split this up in the future, especially when validation comes into play, too. One thing I like in OPA is that we have one test for one thing.